librepo/downloader.c

```diff
@@ -2331,11 +2342,15 @@ check_transfer_statuses(LrDownload *dd, GError **err)
                     target->tried_mirrors = g_slist_remove(target->tried_mirrors, target->mirror);
                     num_of_tried_mirrors = g_slist_length(target->tried_mirrors);
                 }
-
-                if (can_try_more_mirrors(dd, num_of_tried_mirrors))
+                gboolean complete_url_or_baseurl = complete_url_in_path || target->target->baseurl;
```

Comment:
everytime when I see `baseurl` here this function, the code is entirely misleading -- because baseurl isn't only about `baseurl=` statement, but it can mean link to mirrorlist or metalink.

Reply:
Yes I know. Even I believe that baseurl could be completely removed from the code, because it is handled as a single mirror. But it means to rewrite a lot of code.